taskcat/utils.py

```diff
@@ -227,6 +232,22 @@ def get_available_regions(self, service):
         session = boto3.session.Session()
         return session.get_available_regions(service)
 
+    def get_session(self, credential_set, region=None):
+        """fetches existing session for credential set in a region
+
+        Args:
+            credential_set (str): name of credential set from a previously created client
+            region (str): region name, defaults to current region
+
+        Returns:
+            list: aws region name strings
+        """
+        if not region:
+            self.logger.debug("Region not set explicitly, getting default region")
+            region = os.environ['AWS_DEFAULT_REGION']
+
+        return self._clients[credential_set][region]['session']
```

Comment:
Yes, this currently meets the usecase since first the aws_api_init function must be called before the upload one (and it's how the example/cli tool does it). Eventually it would be best to enhance utils properly provision clients and resources based on a set of sessions that it holds as credential sets.